### ISSUE TITLE
Thermostat not reachable

### ISSUE THREAD

**Issue by litinoveweedle** [algirdasc/hass-floureon#5]
Starting from 0.115.0 integration no longer works, as thermostat device is unreachable. I suspect the reason is in my network setup, where thermostat (as well as other smart devices) are in the dedicated network, which is connected to HA network interface as secondary network. This setup caused for example Broadlink based integration to stop working in 0.115.0 as well, due to wrong source IP (of primary network interface) being used for discovery and communication.

More about Broadlink issue [here](https://github.com/home-assistant/core/issues/40267)

Errors in the log (0.116.2):
`2020-10-10 12:44:21 ERROR (MainThread) [custom_components.floureon] Thermostat 192.168.3.57 set_time error: The device is offline
2020-10-10 12:45:20 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading configuration flow for integration broadlink: cannot import name 'NetworkTimeoutError' from 'broadlink.exceptions' (/usr/local/lib/python3.8/site-packages/broadlink/exceptions.py)
2020-10-10 12:45:31 WARNING (MainThread) [custom_components.floureon] Thermostat 192.168.3.57 read_status error: The device is offline`

**Comment by litinoveweedle** [comment]
Based on some suggestion, it would probably help to use [latest python-broadlink](https://github.com/home-assistant/core/pull/39228) in Floureon manifest.

**Comment by felipediel** [comment]
I suggest the adoption of `>=` in the manifest, e.g. `broadlink>=0.15.0`, so this component doesn't prevent users from updating the library in Home Assistant.

**Comment by litinoveweedle** [comment]
Unfortunately only upgrading to 0.116.2 (to provide python-broadlink to 0.15.0) and updating manifest of Floureon integration as suggested didn't fix the issue. Thermostat is still unreachable, it seems that some fix will be need also in integration code.

Anyway I can now confirm, that other Broadlink integration (from HA core) works fine now. So it seems that Floureon integration manifest prevented Broadlink integration to work, which is really surprising to me.

@felipediel Thank you very much for you help and you work on Broadlink itegration.

**Comment by litinoveweedle** [comment]
I think [this](https://github.com/mjg59/python-broadlink/pull/427) (especially point 3.) will have to be considered in Floureon integration to work correctly as well.

currently I am getting:
`2020-10-11 19:32:36 WARNING (MainThread) [custom_components.floureon] Thermostat 192.168.3.57 read_status error: [Errno -4000] Network timeout
2020-10-11 19:33:36 WARNING (MainThread) [custom_components.floureon] Thermostat 192.168.3.57 read_status error: [Errno -4000] Network timeout
2020-10-11 19:34:36 WARNING (MainThread) [custom_components.floureon] Thermostat 192.168.3.57 read_status error: [Errno -4000] Network timeout
2020-10-11 19:35:36 WARNING (MainThread) [custom_components.floureon] Thermostat 192.168.3.57 read_status error: [Errno -4000] Network timeout
2020-10-11 19:36:36 WARNING (MainThread) [custom_components.floureon] Thermostat 192.168.3.57 read_status error: [Errno -4000] Network timeout
2020-10-11 19:37:36 WARNING (MainThread) [custom_components.floureon] Thermostat 192.168.3.57 read_status error: [Errno -4000] Network timeout`

**Comment by algirdasc** [comment]
That is really strange issue, because I'm currently using 0.116.2 with broadlink 0.15.0 and my thermostats (and other broadlink devices) are on separate subnet. Is your separate subnet connected to separate physical interface? Can you ping thermostat from HA machine?

**Comment by algirdasc** [comment]
Ok, I've updated code a bit, now I'm using hello() method to connect to the device. Could you please update and try again? Also, `mac` option in config is deprecated, you can delete it.

**Comment by litinoveweedle** [comment]
Hello, I can confirm that with latest version it definitely works now. Honestly maybe it did work also with previous one, but I just realized, that my HA became very slow, not responding to some integration at all. I do not know, if this was cause or result of previous version of floureon integration which seems to slow system a lot, if thermostat is not reachable. I had to reboot whole HA server and I did rebooted thermostat as well - and now it works OK. I did also removed MAC address as suggested - no issues.

Thank you for your great work, I bought you some coffee, although IMHO beer would be better. ;-)
